services/credential-server/src/modules/signify/signifyApi.ts

```diff
@@ -105,28 +107,49 @@ export class SignifyApi {
     registryId: string,
     schemaId: string,
     recipient: string,
-    attribute: { [key: string]: string }
+    attribute: { [key: string]: any }
   ) {
     await this.resolveOobi(`${config.oobiEndpoint}/oobi/${schemaId}`);
 
     let vcdata = {};
-    if (
-      schemaId === Agent.RARE_EVO_DEMO_SCHEMA_SAID ||
-      schemaId === Agent.QVI_SCHEMA_SAID
-    ) {
+    const schema = await lmdb.get(SCHEMAS_KEY)[schemaId];
+    if (schema) {
       vcdata = attribute;
     } else {
       throw new Error(SignifyApi.UNKNOW_SCHEMA_ID + schemaId);
     }
 
-    const result = await this.signifyClient.credentials().issue(issuerName, {
+    let credentialData = {
       ri: registryId,
       s: schemaId,
       a: {
         i: recipient,
-        ...vcdata,
       },
-    });
+    };
+
+    for (const [key, value] of Object.entries(vcdata)) {
+      const keys = key.split(".");
+      let current = credentialData;
+      for (let i = 0; i < keys.length; i++) {
+        if (i === keys.length - 1) {
+          current[keys[i]] = value;
+        } else {
+          current = current[keys[i]] = current[keys[i]] || {};
```

Comment:
I'm a bit confused on all of this part, mostly because I'm not sure what it looks like. Can you help explain? Why do we split the keys by "." for example

Reply:
As mentioned in the previous comment, with this update to the tool, we’ve moved away from using hardcoded schemas. This means that the inputs on the issuance/request page now generate dynamically.

Initially, I considered storing just the key and its type, which could then be sent to the frontend to populate the form. However, this approach presented a problem: if a key name is used multiple times in the schema, it would create conflicts.

To address this, I decided to store the full path of the key within the schema instead. Now, when the server issues a credential, it looks up the key’s path in the schema and maps it to the data entered in the input fields on the issuance/request form.